ThermalTalk/Common/BasePrinter.cs

```diff
@@ -366,25 +439,47 @@ protected virtual void Dispose(bool diposing)
         /// The port will be closed when the write completes or fails.
         /// </summary>
         /// <param name="payload"></param>
-        protected void internalSend(byte[] payload)
+        /// <returns>ReturnCode.Success if successful, ReturnCode.UnsupportedCommand if payload.Length == 0,
+        /// and ReturnCode.ExecutionFailure otherwise.</returns>
+        protected ReturnCode internalSend(byte[] payload)
         {
             // Do not send empty packets
             if (payload.Length == 0)
             {
-                return;
+                Logger?.Warn("Warning: payload is empty . . .");
+                return ReturnCode.UnsupportedCommand;
             }
+            
+            // for logging
+            var stringData = payload
```

Comment:
This is a lot of linq that will be unused if Logger is null. Add a check if(!(Logger is null)) and then do all this formatting.